client/product_details/AddToBag.jsx

```diff
@@ -39,30 +44,69 @@ class AddToBag extends React.Component {
     });
   }
   
-  handleQuantitySubmit(e) {
+  handleQuantityChange(e) {
     var selectedQuantity = e.target.value;
     this.setState({
       selectedQuantity: selectedQuantity
     });
   }
 
+  handleAddToBagSubmit() {
+    if (this.state.selectedSize !== 'SELECT SIZE') {
+      for (var i = 0; i < this.props.availableSizes.length; i++) {
+        if (this.props.availableSizes[i][0] === this.state.selectedSize) {
+          var sku_id = this.props.availableSizes[i][2];
+          console.log('styleId: ', sku_id);
+        }
+      }
+      var options = {
+        "sku_id": sku_id,
+        "count": this.state.selectedQuantity
+      };
+      // TODO: Add call to post data here & to server
+      fetch(`http://localhost:8080/cart`, {
+        method: 'post',
+        headers: {'Content-Type': 'application/json'},
+        body: JSON.stringify(options)
+      })
+        .then((response) => {
+          console.log(response.status);
+        })
+        .catch((error) => {
+          console.log(error);
+        });
+    } else {
+      // TODO: Add code to open the size dropdown
+      this.setState({
+        showMessage: true
+      });
+    }
+  }
+
   render() {
     var quantityArr = [];
     for (var i = 1; i < (this.state.maxQuantity + 1); i++) {
       quantityArr.push(i);
     }
-    
+    var defaultSizeOption = this.props.availableSizes.length === 0 ? 'OUT OF STOCK' : 'SELECT SIZE';
+    var defaultQuantity = this.state.selectedSize === 'SELECT SIZE' ? '-' : 1;
+
     return (
       <div id="addToBag">
-        <select name={this.state.selectedSize} id="selectSize" placeholder="SELECT SIZE" onChange={this.handleSizeSubmit}>
-          <option value="SELECT SIZE">SELECT SIZE</option>
+        {this.state.showMessage ? <a id="errorMessage">Please Select Size</a> : null}
+        <br></br>
```

Comment:
I also use <br></br> a lot in my component as well, and I'm really curious if it is a better practice to avoid <br> and use CSS (e.g., flexbox) to add some structure to HTML elements instead. Not quite sure myself, but I'm wondering if you have any thoughts on it!

Reply:
Hmm I think at some point we likely should factor those out. But that would mean that we need to put each element in a div and set some sort of margin and padding. I wanted to avoid that for now but definitely agree we should consider that while doing some cleanup